### PR TITLE
Tickets/dm36802 - Convert ts_xml to pyproject.toml

### PR DESCRIPTION
**Pull request by rbovill**
* Moved sal_interfaces/ and schema/ directories under python/lsst/ts/xml/data
* Updated Python modules and unit tests, accordingly.
* Added pyproject.toml, removed requirements.txt and updated setup.cfg and conda/meta.yaml.

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{% set data= load_setup_py_data() %}
+package: 
+  name: ts-integrationtests
```
Comment (tribeiro) [inline conda/meta.yaml:3]: This should probably be `ts-xml`

File: conda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  version: {{ data.get('version') }}
+
+source:
+  path: ..
+
+build:
+  noarch: generic
+  script: python -m pip install --ignore-installed -e .
```
Comment (tribeiro) [inline conda/meta.yaml:11]: Take a look at [this confluence page](https://confluence.lsstcorp.org/x/0Il-Cw), it has some updates to the conda recipe to make them forward compatible with different versions of python..

File: python/lsst/ts/xml/generate_csv_table.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def generate_csv_table():
     tree = ElementTree.parse(sal_interfaces_dir / "SALSubsystems.xml")
     root = tree.getroot()
 
-    outpath = utils.get_pkg_root() / "doc" / "subsystemData.csv"
+    outpath = utils.get_pkg_root() / "subsystemData.csv"
```
Comment (wvreeven) [inline python/lsst/ts/xml/generate_csv_table.py:35]: I still very strongly think that the `subsystemData.csv` file should not be moved from the `doc` directory. Are you sure that this is absolutely necessary? I don't think that any changes to the `generate_csv_table.py` file are necessary, including the change to line 141.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Sorry for pushing back on the changes made to one python file. I really don't think those are necessary and I fear they will still break the way the Makefile in the doc directory has been implemented.

**Review by r-owen — APPROVED** [review]
I have one suggested change and I'll leave it to you and @wvreeven to decide what to do about that one generated file.

File: setup.cfg [hunk 4]
```diff
@@ -1,18 +1,9 @@
...
 max-line-length = 110
 max-doc-length = 79
 ignore = E133, E226, E228, N802, N803, N806, N812, N813, N815, N816, W503
 exclude =
-  sal_interfaces/*,
-  schema/*,
-  scripts/*,
-  tests/.tests,
+  doc,
+  doc/conf.py,
```
Comment (r-owen) [inline setup.cfg:7]: I gently suggest making doc/conf.py pass flake8. It makes more a much more pleasant experience when editing the file in some editors. It does require some # noqa entries. Here is an example from ts_utils:
```
from documenteer.conf.pipelinespkg import *  # type: ignore # noqa
import lsst.ts.utils  # noqa

project = "ts_utils"
html_theme_options["logotext"] = project  # type: ignore # noqa
html_title = project
html_short_title = project
# Avoid warning: Could not find tag file _doxygen/doxygen.tag
doxylink = {}  # type: ignore
```

**Review by r-owen — APPROVED** [review]
Looks fine to me

File: conda/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+source:
+  path: ..
+
+build:
+  noarch: generic
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed .
+
+test:
+  requires:
+    - astropy
```
Comment (tribeiro) [inline conda/meta.yaml:15]: Please, include 

```
    - ts-conda-build =0.3
```

here.

File: conda/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    - python {{ python }}
+    - pip
+    - setuptools_scm
+    - setuptools
+  build:
+    - python {{ python }}
+    - setuptools_scm
+    - setuptools
+  run:
+    - python {{ python }}
```
Comment (tribeiro) [inline conda/meta.yaml:36]: I think you also need `astropy` and `lxml` as run dependencies (but no `ts-conda-build`, which is only a test dependency).

**Review by tribeiro — APPROVED** [review]
Looks good! I did left a few more comments on the conda recipe! please, take a look before merging. The current recipe won't build properly because it is missing a test dependency and it won't run properly because it is missing some run dependencies.

**Review by wvreeven — APPROVED** [review]
Thanks for the additional changes. Looks good to me now.

**Pull request merged**